### ISSUE TITLE
Possible bug on ARM

### ISSUE THREAD

**Issue by whmountains** [gnzlbg/slice_deque#64]
I have a [data collection daemon](https://gitlab.com/szaver/sensorpi) running on several Raspberry Pis which crash regularly (every few days) with the following error:

```
*** Error in `./sensorpi': free(): invalid pointer: 0x76b65f01 ***
./run.sh: line 3:     8 Aborted                 (core dumped) ./sensorpi
```

My crate doesn't have any unsafe code, so it must be an issue with the dependencies.  I believe it is either in `slice_deque` or `gnuplot`.  I am going to try replacing them one at a time and see if the error goes away.  I will keep you updated and close this issue if it turns out not to be an issue with this crate.

Maybe we should run some fuzzing on Arm as well as x86?

**Comment by gnzlbg** [comment]
You might be hitting #57 , if it turns out to be a `slice_deque` issue, it would be awesome if you could reduce an example that we could use to fix this.

**Comment by gnzlbg** [comment]
@whmountains do you still have this issue or was it fixed by #57 ?

**Comment by whmountains** [comment]
Seems to have been fixed by #57.
